VideoAPI/Video.API/Controllers/ConsultationController.cs

```diff
@@ -240,6 +246,55 @@ public async Task<IActionResult> StartPrivateConsultationWithEndpointAsync(Endpo
 
             return Accepted();
         }
+
+        [HttpPost("/consultations/start")]
+        [SwaggerOperation(OperationId = "StartPrivateConsultation")]
+        [ProducesResponseType((int) HttpStatusCode.Accepted)]
+        [ProducesResponseType((int) HttpStatusCode.NotFound)]
+        [ProducesResponseType((int) HttpStatusCode.Unauthorized)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [AllowAnonymous]
+        public async Task<IActionResult> StartConsultationRequestAsync(StartConsultationRequest request)
+        {
+            _logger.LogDebug("StartJudgeJohConsultationRequest");
+            Room room;
+            Participant participant;
+            
+            try
+            {
+                var query = new GetAvailableRoomByRoomTypeQuery(request.RoomType, request.ConferenceId);
+                var listOfRooms = await _queryHandler.Handle<GetAvailableRoomByRoomTypeQuery, List<Room>>(query);
+                room = listOfRooms.FirstOrDefault(x => x.Type.Equals(request.RoomType));
+                if (room == null)
```

Comment:
It's quite busy here in this controller. Could you refactor these steps out into methods please like we spoke about. Thanks!